extensions/amp-analytics/0.1/variables.js

```diff
@@ -276,6 +296,21 @@ export class VariableService {
     this.macros_[name] = macro;
   }
 
+  /**
+   * Registers a macro, but disallows access in FIEs
+   * @param {string} name
+   * @param {*} macro
+   */
+  registerFieDisallowed_(name, macro) {
+    const element = this.ampdoc_.win.document.documentElement;
```

Comment:
FIE and the parent doc share the same ampdoc. the element here will always be the parent doc's  documentElement. You'll need to register the macro in `getMacros` method based on the element.